### PR TITLE
DEPR: string indexing along index for datetimes

### LINKED ISSUE #31476: DEPR: DataFrame.__getitem__[str] sometimes slices on index
xref #31334, #9595 

- This is a tiny corner case, at least as measured by tests cases (2 tests reach this)
- With this removed, the `__getitem__` API becomes much simpler to describe: "DataFrame.__getitem__` is always column-based"

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #31476
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Let's get ourselves one less special case to worry about

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
on board with this, ping when passing

**Review by jreback — CHANGES_REQUESTED** [review]
also the ???

**Comment by jbrockmendel** [comment]
> also the ???

placeholder before I had a PR number.  just updated

**Comment by jbrockmendel** [comment]
could use some help with the ipython directive errors

**Comment by jbrockmendel** [comment]
ping

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks, can you add the entry to the deprecation removal issue as well when you have a sec

File: doc/source/user_guide/timeseries.rst [hunk 1]
```diff
@@ -590,24 +595,28 @@ This starts on the very first time in the month, and includes the last date and
 time for the month:
 
 .. ipython:: python
+   :okwarning:
 
    dft['2013-1':'2013-2']
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/timeseries.rst:600]: This is not actually deprecated, right? (and thus shouldn't raise a warning?)

Comment (jbrockmendel) [reply doc/source/user_guide/timeseries.rst:600]: Good catch.  I think this should be deprecated too.

**Pull request merged**